bConnect/Public/Edit-bConnectJob.ps1

```diff
@@ -0,0 +1,25 @@
+Function Edit-bConnectJob() {
+    <#
+        .Synopsis
+            Updates a existing job.
+        .Parameter Job
+            Job object (hashtable).
+        .Outputs
+            Job (see bConnect documentation for more details).
+    #>
+
+
+    Param (
+        [Parameter(Mandatory=$true)][PSCustomObject]$Job,
+        [boolean]$IgnoreAssignments = $false
+
+    )
+
+    $_connectVersion = Get-bConnectVersion
+    If($_connectVersion -ge "1.0") {
+        $_Job = ConvertTo-Hashtable $Job
+        return Invoke-bConnectPatch -Controller "Jobs" -Version $_connectVersion -objectGuid $_Job.Id -Data $_Job -IgnoreAssignments $IgnoreAssignments
```

Comment:
maybe add Test-Guid
as in: 
https://github.com/baramundisoftware/PS-bConnect/blob/6f8b4c1abf4de1b0b2f42fa354b588c4f846d229/bConnect/Public/Edit-bConnectDynamicGroup.ps1#L18
https://github.com/baramundisoftware/PS-bConnect/blob/6f8b4c1abf4de1b0b2f42fa354b588c4f846d229/bConnect/Public/Edit-bConnectEndpoint.ps1#L18
https://github.com/baramundisoftware/PS-bConnect/blob/6f8b4c1abf4de1b0b2f42fa354b588c4f846d229/bConnect/Public/Edit-bConnectOrgUnit.ps1#L18
https://github.com/baramundisoftware/PS-bConnect/blob/6f8b4c1abf4de1b0b2f42fa354b588c4f846d229/bConnect/Public/Edit-bConnectStaticGroup.ps1#L18
https://github.com/baramundisoftware/PS-bConnect/blob/6f8b4c1abf4de1b0b2f42fa354b588c4f846d229/bConnect/Public/Edit-bConnectVariableDefinition.ps1#L18